server/bin/pbench-server-prep-shim-002.py

```diff
@@ -110,147 +79,41 @@ def process_tb(config, logger, receive_dir, qdir_md5, duplicates, errors):
-            logger.info("{}: FAILED", tb.name)
-            logger.info("md5sum: WARNING: 1 computed checksum did NOT match")
-            quarantine((qdir_md5 / controller), logger, tb, tbmd5)
-            nquarantined += 1
-            continue
-
-        if dataset:
-            try:
-                dataset.md5 = archive_md5_hex_value
-                dataset.update()
-            except DatasetError as e:
-                logger.warn(
-                    "Unable to update dataset {} with md5: {}", str(dataset), str(e)
-                )
-
-        # make the destination directory and its TODO subdir if necessary.
         try:
-            os.makedirs(dest / "TODO")
-        except FileExistsError:
-            # directory already exists, ignore
-            pass
-        except Exception:
-            logger.error("{}: Error in creating TODO directory.", config.TS)
-            quarantine(os.path.join(errors, controller), logger, tb, tbmd5)
+            move_results.push(controller, str(tb))
+        except FileNotFoundError as e:
+            logger.error("File Not Found error, {}", e)
             nerrs += 1
             continue
-
-        # First, copy the small .md5 file to the destination. That way, if
-        # that operation fails it will fail quickly since the file is small.
-        try:
-            shutil.copy2(tbmd5, dest)
-        except Exception:
-            logger.error(
-                "{}: Error in copying .md5 file to Destination path.", config.TS
-            )
-            try:
-                os.remove(dest / tbmd5.name)
-            except FileNotFoundError:
-                logger.error(
-                    "{}: Warning: cleanup of copy failure failed itself.", config.TS
-                )
-            quarantine((errors / controller), logger, tb, tbmd5)
+        except (FileUploadError, RuntimeError) as e:
+            logger.error("Error uploading a file, {}", e)
             nerrs += 1
             continue
-
-        # Next, mv the "large" tar ball to the destination. If the destination
-        # is on the same device, the move should be quick. If the destination is
-        # on a different device, the move will be a copy and delete, and will
-        # take a bit longer.  If it fails, the file will NOT be at the
-        # destination.
-        try:
-            shutil.move(str(tb), str(dest))
-        except Exception:
-            logger.error(
-                "{}: Error in moving tarball file to Destination path.", config.TS
-            )
-            try:
-                os.remove(dest / resultname)
-            except FileNotFoundError:
-                logger.error(
-                    "{}: Warning: cleanup of copy failure failed itself.", config.TS
-                )
-            quarantine((errors / controller), logger, tb, tbmd5)
+        except Exception as e:
+            logger.error("Exception === ", e)
             nerrs += 1
             continue
```

Comment:
Since the action taken in each of the exception handlers is substantially the same (log a message, count the error, and continue with the next iteration of the loop), there's no particular need to have multiple handlers -- I believe that the stringization of the exception object should provide a sufficient message in each case.

Reply:
ok